### ISSUE TITLE
ValueError when trying to Trash temporary Page created generated from hash symbol `page#something`

### ISSUE THREAD

**Issue by hsandt** [zim-desktop-wiki/zim-desktop-wiki#2384]
**Describe the bug**
When you enter # in a string, Zim auto-creates a temporary page out of side. It happens a lot with Discord usernames (fortunately the hash will soon disappear) and there is no way to disable this. Generally I can still right-click and delete the temp page OR right-click on the link and Remove link so I don't get bothered by the ghost page.

But this time, trying to delete the temp page caused an error:

Looks like we found a bug
When reporting this bug please include the information from the text box below

> This is zim 0.75.1
Platform: posix
Locale: en_US UTF-8
FS encoding: utf-8
Python: (3, 10, 6, 'final', 0)
PyGObject: (3, 42, 1)

> ======= Traceback =======
  File "/usr/lib/python3/dist-packages/zim/gui/widgets.py", line 3084, in do_response
    destroy = self.do_response_ok()
  File "/usr/lib/python3/dist-packages/zim/gui/uiactions.py", line 762, in do_response_ok
    raise op.exception
  File "/usr/lib/python3/dist-packages/zim/notebook/operations.py", line 243, in <lambda>
    GObject.idle_add(lambda: next(my_iter, False), priority=GObject.PRIORITY_LOW)
  File "/usr/lib/python3/dist-packages/zim/notebook/operations.py", line 270, in __iter__
    progress = next(self._do_work)
  File "/usr/lib/python3/dist-packages/zim/notebook/notebook.py", line 898, in trash_page_iter
    for p in self._deleted_page(path, update_links):
  File "/usr/lib/python3/dist-packages/zim/notebook/notebook.py", line 945, in _deleted_page
    self._remove_links_in_page(page, path)
  File "/usr/lib/python3/dist-packages/zim/notebook/notebook.py", line 972, in _remove_links_in_page
    newtree = tree.substitute_elements((zim.formats.LINK,), replacefunc)
  File "/usr/lib/python3/dist-packages/zim/formats/__init__.py", line 638, in substitute_elements
    replacement = func(TokenListElement(t[0], t[1], content))
  File "/usr/lib/python3/dist-packages/zim/notebook/notebook.py", line 963, in replacefunc
    hrefpath = self.pages.lookup_from_user_input(href, page)
  File "/usr/lib/python3/dist-packages/zim/notebook/index/pages.py", line 734, in lookup_from_user_input
    href = HRef.new_from_wiki_link(name)
  File "/usr/lib/python3/dist-packages/zim/notebook/page.py", line 375, in new_from_wiki_link
    names = Path.makeValidPageName(href.lstrip('+')) if href else ""
  File "/usr/lib/python3/dist-packages/zim/notebook/page.py", line 155, in makeValidPageName
    raise ValueError('Not a valid page name: %s (was: %s)' % (newname, name))
ValueError: Not a valid page name:  (was: ")

**To Reproduce**
1. Paste a string like "Username#3587" and press Space if needed
2. Note how string becomes hypelinked in blue and a ghost page "Username" appears
3. Right-click on page > Delete Page
4. Confirm with OK (you can keep remove link checkbox checked)
5. See error. You must close the popup and Cancel the op. Deleting failed.

**Expected behavior**
The page should be deleted without error.

**Screenshots**

![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/1067772/93a74d87-6fc0-4716-85e7-cd962fac866f)
![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/1067772/8b5fc11d-6c2d-4d64-9143-4d83a4fcf404)

**Environment (please complete the following information):**
 - Version 0.75.1 (Ubuntu PPA)
 - OS: Linux Ubuntu 22.04
 - Language: English

**Additional context**
It just happened today, with 2 generated ghost pages in a row... However it didn't happen previously. So I can't guarantee you'll repro it.

Right-click > Remove Link _does_ work so that's a good workaround for now.

**Comment by hsandt** [comment]
I noticed that temp page "Username" is placed at the same level (sibling) as the page you wrote Username#... inside, instead of being a child as I'd expect. Maybe it's related? Can't remember what it was doing before.

**Comment by introt** [comment]
Can't reproduce on latest develop. This seems to be another case of anchor links being funky. See also: #1942

> and there is no way to disable this

There is: untick Preferences > Automatically turn wiki page names into links

> I noticed that temp page "Username" is placed at the same level (sibling) as the page you wrote Username#... inside

This is expected, as "Username:Page" also places "Username" as a sibling. It'd appear as a child if it started with +

**Comment by hsandt** [comment]
Thanks, it's working now:

Repro on v0.75.2: cannot repro all steps anymore because now, after pressing space, the page will not immediately be created. I need an extra step between 2 and 3 to click on another page to make the ghost greye out page appear. From here, I can right-click delete it without error.

I unticked Preferences > Automatically turn wiki page names into links and also ... identifiers starting with "#" into links so I'm not bothered by a#b and #a altogether.

I tried +a#b and it does as you say. Although I won't be typing it myself and just copy link to other pages after creating them only.
